gateway/src/main/java/io/zeebe/gateway/impl/probes/health/ResponsiveHealthIndicator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.impl.probes.health;
+
+import static java.util.Objects.requireNonNull;
+
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.ZeebeClientBuilder;
+import io.zeebe.gateway.impl.configuration.GatewayCfg;
+import java.time.Duration;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Health.Builder;
+import org.springframework.boot.actuate.health.HealthIndicator;
+
+/**
+ * Health indicator that sends a request to the gateway to check its responsiveness. The request
+ * must yield a response within a given timeout for this health indicator to report {@code
+ * Status.UP}
+ */
+public class ResponsiveHealthIndicator implements HealthIndicator {
```

Comment:
Will be happy to discuss this.

My rationales were as follows:
* Topology is the only request I know of that has no side effects
* When researching best practices for health check one recommendation was to make a local health check, and not an aggregate health check of all the stuff you depend upon. As such, a request that is relayed to a broker is not ideal
* As part of the failure case documentation we wanted to make sure that
  * there is no deadlock in the Gateway request processing thread
  * the Gatewa is not so overwhelmed with requests that it cannot accept any additional requests
* Both these situation would be detected by this health check as it is implemented now

Reply:
> Both these situation would be detected by this health check as it is implemented now

Is this really the case? Because what I think what we miss here is the transport part to the broker.

I created an example see below:

![Gateway Overview](https://user-images.githubusercontent.com/2758593/82811163-64b0a980-9e90-11ea-8a85-52d8cc5e7550.png)

We just check that the point to endpoint and topology manager works, but not the direction to BrokerClient and Broker works. Do you get what I mean?

Reply:
Sure, but the gateway would be healthy even if the network (to the broker) is down. The gateway is responsive, it is not blocked by deadlock and not overwhelmed by requests.

We could add another health check, whether the Gateway ca reach all brokers, but I am not sure it's a good idea.

Happy to discuss it though.

Reply:
Hm, I somehow have the feeling that we miss a part. What I thought is that it might be possible to answer the topology request but the broker client component is somehow overhelmed because of so many requests. But yeah maybe the current approach is fine for the first step.